### PR TITLE
Pal 665 readers added JavaDoc where missing

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java [hunk 1]
```diff
@@ -53,10 +53,21 @@
     @JsonIgnore
     private FileSystem fs;
 
+    /**
+     * Creates a new {@link HadoopDataReader} object
+     *
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java:59]: Why would this exception be thrown?

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java [hunk 2]
```diff
@@ -53,10 +53,21 @@
...
+     */
     public HadoopDataReader() throws IOException {
         conf(new Configuration());
     }
 
+    /**
+     * Creates a new {@link HadoopDataReader} object
+     *
+     * @param conf          A {@link Map} of {@link String}s used as the configuration
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java:69]: ditto

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java [hunk 3]
```diff
@@ -72,17 +83,36 @@ private static Configuration createConfig(final Map<String, String> conf) {
         return config;
     }
 
+    /**
+     *
+     * @param conf          A {@link Map} of {@link String}s used as the configuration
+     * @return              the current {@link HadoopDataReader} object
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java:90]: ditto

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java [hunk 4]
```diff
@@ -72,17 +83,36 @@ private static Configuration createConfig(final Map<String, String> conf) {
...
     public HadoopDataReader conf(final Map<String, String> conf) throws IOException {
         return conf(createConfig(conf));
     }
 
+    /**
+     * Sets the {@link FileSystem} value
+     *
+     * @param conf          A Hadoop {@link Configuration} object
+     * @return              the current {@link HadoopDataReader} object
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java:102]: ditto

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 5]
```diff
@@ -72,12 +72,24 @@
 
     private List<ConnectionDetail> dataServices = new ArrayList<>();
 
+    /**
+     * Creates a new {@link HadoopResourceService} object
+     *
+     * @param config        A Hadoop {@link Configuration} object
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java:79]: ditto

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 6]
```diff
@@ -72,12 +72,24 @@
...
+     */
     public HadoopResourceService(final Configuration config) throws IOException {
         requireNonNull(config, "service");
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java:82]: this probably shouldn't be saying "service" anymore

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java [hunk 7]
```diff
@@ -72,17 +83,36 @@ private static Configuration createConfig(final Map<String, String> conf) {
         return config;
     }
 
+    /**
+     *
```
Comment (developer01189998819991197253) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/reader/HadoopDataReader.java:87]: needs a descripting line

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 8]
```diff
@@ -72,12 +72,24 @@
...
         this.config = config;
         this.fileSystem = FileSystem.get(config);
     }
 
+    /**
+     * Creates a new {@link HadoopResourceService} object
+     *
+     * @param conf          A {@link Map} of {@link String}s used as the configuration
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java:91]: ditto

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 9]
```diff
@@ -227,6 +239,13 @@ protected LeafResource addConnectionDetail(final HadoopResourceDetails hadoopRes
                 .connectionDetail(dataService);
     }
 
+    /**
+     * Sets the {@link Configuration} and {@link FileSystem} values
+     *
+     * @param conf          A Hadoop {@link Configuration} object
+     * @return              the current {@link HadoopResourceService} object
+     * @throws IOException  the {@link Exception} thrown when there is an issue
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java:247]: ditto

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java [hunk 10]
```diff
@@ -68,6 +75,12 @@ public static void addTypeSupport(final String type, final String classString) {
         SUPPORTED_TYPES.put(type, classString);
     }
 
+    /**
+     * Creates a new {@link HadoopResourceDetails} object using the file name
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java:79]: How is this filename used? What sort of format is expected?

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java [hunk 11]
```diff
@@ -88,6 +101,12 @@ public static HadoopResourceDetails getResourceDetailsFromFileName(final URI fil
 
     }
 
+    /**
+     * Checks that the file name is valid
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java:105]: Valid as to what validation check?

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 12]
```diff
@@ -72,12 +72,24 @@
 
     private List<ConnectionDetail> dataServices = new ArrayList<>();
 
+    /**
+     * Creates a new {@link HadoopResourceService} object
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java:76]: needs a bit more desciption, i.e what it sets the object from

File: readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/DataReader.java [hunk 13]
```diff
@@ -65,13 +65,29 @@
...
+    /**
+     * Adds a serialiser with the {@link DataFlavour} and {@link Serialiser} values
+     *
+     * @param flavour       the {@link DataFlavour} value to be added
+     * @param serialiser    the {@link Serialiser} value to be added
+     */
     void addSerialiser(final DataFlavour flavour, final Serialiser<?> serialiser);
 
+    /**
+     * Gets the class name
```
Comment (dev930018) [inline readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/DataReader.java:77]: This used to be used for json things, but data-readers never get marshalled to json, so why is it here at all?

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java [hunk 14]
```diff
@@ -49,6 +49,13 @@
     private String type;
     private String format;
 
+    /**
+     * Constructs a new {@link HadoopResourceDetails} object
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java:53]: needs a bit more description

File: readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/SerialisedDataReader.java [hunk 15]
```diff
@@ -134,6 +149,13 @@ public DataReaderResponse read(final DataReaderRequest request, final AtomicLong
         return (Serialiser<T>) serialiser;
     }
 
+    /**
+     * Gets the associated {@link Serialiser} from the {@link Map} for a {@link LeafResource}.
+     *
+     * @param resource  the {@link LeafResource} used to get the {@link Serialiser}
+     * @param <I>       the type of {@link Serialiser}
+     * @return          the associatedd {@link Serialiser}
```
Comment (dev930018) [inline readers-common/src/main/java/uk/gov/gchq/palisade/reader/common/SerialisedDataReader.java:157]: typo

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java (hunk 11 above)
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/util/HadoopResourceDetails.java:105]: needs a bit more description, such as what it returns if valid, and if invalid, and an example of valid vs invalid

File: readers-common/src/main/java/uk/gov/gchq/palisade/reader/request/DataReaderRequest.java [hunk 16]
```diff
@@ -43,6 +43,8 @@ public DataReaderRequest() {
     }
 
     /**
+     * Set the resource object for this request
```
Comment (developer01189998819991197253) [inline readers-common/src/main/java/uk/gov/gchq/palisade/reader/request/DataReaderRequest.java:46]: needs a bit more description

File: readers-common/src/main/java/uk/gov/gchq/palisade/reader/request/DataReaderResponse.java [hunk 17]
```diff
@@ -50,6 +50,8 @@ public DataReaderResponse writer(final ResponseWriter writer) {
     }
 
     /**
+     * Set the message value for this response
```
Comment (developer01189998819991197253) [inline readers-common/src/main/java/uk/gov/gchq/palisade/reader/request/DataReaderResponse.java:53]: needs a bit more description

**Pull request merged**